multy_core/src/bitcoin/bitcoin_account.cpp

```diff
@@ -117,6 +117,65 @@ class BitcoinP2PKHAccount : public BitcoinAccount
     }
 };
 
+class BitcoinP2SHWPKHAccount : public BitcoinAccount
+{
+public:
+    using BitcoinAccount::BitcoinAccount;
+
+    std::string get_address() const override
+    {
+
+        // P2SHPKH address generated from public key.
+        unsigned char pub_hash[HASH160_LEN + 1] = {'\0'};
+
+        // 1 - Take the corresponding public key generated with it (33 or 65
+        // bytes)
+        PublicKeyPtr public_key(m_private_key->make_public_key());
+        BinaryData key_data = public_key->get_content();
+
+        // 2 - Perform SHA-256 hashing on the public key
+        // 3 - Perform RIPEMD-160 hashing on the result of SHA-256
+        // 4 - Perform make script segWit for lock bitcoins
+        // 5 - Perform SHA-256 hashing on the segWit script
+        // 6 - Perform RIPEMD-160 hashing on the result of SHA-256
+
+        {
+            unsigned char segwit_script[HASH160_LEN + 2] = {'\0'};
+           // Leave the first two bytes intact for script opcode
+            BinaryData hash_data_1 = power_slice(as_binary_data(segwit_script), 2, -2);
+            bitcoin_hash_160(key_data, &hash_data_1);
+            segwit_script[0] = 0x00; // OP_0
```

Comment:
There is an `OP_CODE enum` in bitcoin_transaction.cpp, you should use that instead of hardcoding opcodes.